### ISSUE TITLE
Eval ReAct

### ISSUE THREAD

**Issue by JasonLo** [UW-Madison-DSI/ask-xDD#80]
[ReAct](https://react-lm.github.io/) specializes in breaking down complex questions into simpler sub-questions and conducts related actions (e.g. searches) to provide answers. It performs well on queries like "Compare model A and B," where it might splits the question into "What is model A," "What is model B," and then compares the results.

To evaluate ReAct's performance, I selected 10 complex queries from the [TA1-testset](https://docs.google.com/spreadsheets/d/1M55Fx4Sr_FpWZ0LjtZCXJecUvTaql2LkrOovNndwacw/edit#gid=297881431). 

- Baseline (v02): hybrid retriever + original summarizer
- Experimental (v03): hybrid retriever + ReAct

tl;dr: The evaluation showed a modest improvement in response quality by successfully breaking down complex queries.

Potential Improvements:

1. Integrate additional tools into ReAct, such as a formula object for formula-related queries.
2. Enhance the retriever using features like "more_like" and "less_like" to improve search results.

Full evaluation details are [here](https://docs.google.com/spreadsheets/d/1uy38vAhCB7qik0x-DEjpkFXvWG2lsLozxTBfQHNUTLk/edit#gid=0).
